themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,136 @@
+        ...
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+        ...
+	})
+    ```
+
+    To set the trigger in your serverless function, navigate to your `function.json` configuration and set the `type` parameter in the `bindings` object.
+
+    ``` json
+    "bindings": [
+        {
+            "authLevel": "Anonymous",
+            "type": "httpTrigger",
+            "direction": "in",
+            "name": "req",
+        "methods": [
+            "get"
+        ]
+        },
+        {
+            "type": "http",
+            "direction": "out",
+            "name": "res"
+        }
+    ]
+
+    ```
+
+- **Scalability**: Serverless doesn't mean "no server" otherwise, where will the code actually run? The meaning of serverless is "less server." When event density rises, the majority of serverless platforms offer a set of options to manage how the infrastructure should scale. There are several options available to you, and the technique you use will depend on the function. Azure Functions offer auto-scaling, allowing your application to seamlessly handle increased workloads.  By designing your serverless functions with scalability in mind, you can maintain optimal performance even during traffic spikes, delivering a responsive and reliable experience to your users.
+
+- **Dependency across multiple services**: In your serverless architecture, there are times when functions depend on other functions. This typically happens a lot in distributed systems. For example in an e-commerce service, the checkout function could depend on the payment function. However, it's advised that services avoid making direct references to one another in order to prevent hard coupling that can cause issues later. A significant amount of reworking may be required if direct references for a service need to be changed. Using message services, such as queues or topics, to facilitate service-to-service communication is a good way to handle this.
+
+> Figure 1. Resources needed in a serverless architecture
+![serverless-resources](./serverless-resources.png)
+
+To create serverless infrastructure for an application that should have distint function logic, statelessness, message queue triggers, auto scaling, and service to service communication, you can identify all the cloud resources needed for your infrastructure and use Pulumi to declare and manage those resources. The interesting thing about Pulumi's code-driven infrastructure provisioning is that you can manage all your serverless infrastructure in one place which makes development faster, and treating IaC code like regular code also gives you the ability to track changes to infrastructure over time.
+
+If you've joined us on this incredible journey to lean about serverless applications with Pulumi and Azure, we'd love to hear your story. How did it work out for you?
```

Comment:
```suggestion
If you've joined us on this incredible journey to learn about serverless applications, we'd love to hear your story. How did it work out for you?
```

Reply:
"incredible journey", the voice is a bit off from how we want to speak with our customer. it needs to be a bit more down to earth.